confgenerator/logging_receivers.go

```diff
@@ -152,6 +152,53 @@ func init() {
 	loggingReceiverTypes.registerType(func() component { return &LoggingReceiverSyslog{} })
 }
 
+// A LoggingReceiverTCP represents the configuration for a TCP receiver.
+type LoggingReceiverTCP struct {
+	ConfigComponent `yaml:",inline"`
+
+	ListenHost string `yaml:"listen_host,omitempty" validate:"omitempty,ip"`
+	ListenPort uint16 `yaml:"listen_port,omitempty"`
+}
+
+func (r LoggingReceiverTCP) Type() string {
+	return "tcp"
+}
+
+func (r LoggingReceiverTCP) Components(tag string) []fluentbit.Component {
+	if r.ListenHost == "" {
+		r.ListenHost = "0.0.0.0"
```

Comment:
Had some offline discussion about the default value here.

google-fluentd defaults to `127.0.0.1`: https://github.com/GoogleCloudPlatform/fluentd-catch-all-config/blob/bf6ff8ee98a0d0fedf88f2e95113fc6c172cc986/configs/config.d/syslog_endpoint.conf#L5

We'd like to ensure the default / out of the box experience limits the access as much as possible.